test/blackbox/tree/TreeTests.cpp

```diff
@@ -267,6 +267,676 @@ TEST_F(TreeTests, REFTree)
+                                                 "<historyQos>"
+                                                     "<kind>KEEP_LAST</kind>"
+                                                     "<depth>10</depth>"
+                                                 "</historyQos>"
+                                                 "<durability>"
+                                                     "<kind>TRANSIENT_LOCAL</kind>"
+                                                 "</durability>"
+                                             "</topic>"
+                                         "</data_reader>"
+                                     "</dds>";
+
+    /* Create client. */
+    dds::xrce::AGENT_Representation agent_representation;
+    dds::xrce::CLIENT_Representation client_representation;
+    client_representation.xrce_cookie(dds::xrce::XRCE_COOKIE);
+    client_representation.xrce_version(dds::xrce::XRCE_VERSION);
+    client_representation.xrce_vendor_id(vendor_id_);
+    client_representation.client_timestamp().seconds(0x00);
+    client_representation.client_timestamp().nanoseconds(0x00);
+    client_representation.client_key(client_key_);
+    client_representation.session_id(0x00);
+    dds::xrce::ResultStatus response = root_.create_client(client_representation, agent_representation);
+    std::shared_ptr<ProxyClient> client = root_.get_client(client_representation.client_key());
+
+    /* Common creation mode. */
+    dds::xrce::CreationMode creation_mode;
+
+    /*
+     * Create participant.
+     */
+    dds::xrce::ObjectVariant object_variant;
+    dds::xrce::OBJK_PARTICIPANT_Representation participant_representation;
+    participant_representation.domain_id(0);
+    participant_representation.representation().xml_string_representation(participant_xml);
+    object_variant.participant(participant_representation);
+
+    dds::xrce::ObjectPrefix participant_id = {0x00, 0x01};
+
+    creation_mode.reuse(false);
+    creation_mode.replace(false);
+    response = client->create(creation_mode, participant_id, object_variant);
+    ASSERT_EQ(dds::xrce::STATUS_OK, response.status());
+
+    response = client->delete_object(participant_id);
+    creation_mode.reuse(false);
+    creation_mode.replace(true);
+    response = client->create(creation_mode, participant_id, object_variant);
+    ASSERT_EQ(dds::xrce::STATUS_OK, response.status());
+
+    response = client->delete_object(participant_id);
+    creation_mode.reuse(true);
+    creation_mode.replace(false);
+    response = client->create(creation_mode, participant_id, object_variant);
+    ASSERT_EQ(dds::xrce::STATUS_OK, response.status());
+
+    response = client->delete_object(participant_id);
+    creation_mode.reuse(true);
+    creation_mode.replace(true);
+    response = client->create(creation_mode, participant_id, object_variant);
+    ASSERT_EQ(dds::xrce::STATUS_OK, response.status());
+
+    creation_mode.reuse(false);
```

Comment:
From here maybe split in multiple tests.

Reply:
It will be simplify for the next release.